source/_integrations/pid_controller.markdown

```diff
@@ -0,0 +1,152 @@
+---
+title: PID Controller
+description: Creates a PID controller to regulate a sensor value with a number-entity output.
+ha_category:
+  - Number
+ha_release: '2023.05'
+ha_iot_class: Local Polling
+ha_domain: pid_controller
+ha_platforms:
+  - number
+ha_integration_type: integration
+---
+
+The `pid_controller` platform is a `number`. It can be used to control a number entity output to regulate a sensor value to a certain setpoint. The value of the number entity is the setpoint. As a sensor, any numerical senosor entity can be used. If two sensors are configured, the PID controller will act as a differential controller, using the difference between the two sensor values as input signal.
+The value for the output number entity will be calculated using the Proportional–Integral–Derivative algorithm (PID, See [https://en.wikipedia.org/wiki/PID_controller]). The implementation of the PID controller contains bumpless operation, and is prevented against integral windup by clipping of the output value to the minimum and maximum of the corresponding output number entity. 
+This controller is typically usefull in regulated systems, for example regulate the speed of a water pump in a heat collector to keep the temperature difference between the outgoing and incomming waterstream on a certain level, so that the heat collector will perform optimal.
```

Comment:
```suggestion
This controller is typically useful in regulated systems. For example, to regulate the speed of a water pump in a heat collector to keep the temperature difference between the outgoing and incoming water stream on a certain level, so that the heat collector will perform optimally.
```